notification/logic/PostLogic.java

```diff
@@ -0,0 +1,163 @@
+	    }
+	    
+    	response.setTime(LocalDateTime.now().toString());
+    	response.setToken(body.getToken());
+    	response.setId(body.getId());
+    	
+    	return Response.status(Response.Status.CREATED.getStatusCode()).entity(response).build();
+		
+	}
+	
+	private Response bodyTypeCheck(RequestBody body) {
+		
+		Responses response = new Responses();
+		
+	    if (stringutil.isNullorEmpty(body.getId()) || stringutil.isNullorEmpty(body.getToken())) {
+	    
+	    	return  GeneralErrorRes(ErrorResponseEnum.E400);
+	    }
+	  
+	    return Response.status(Response.Status.CREATED.getStatusCode()).entity(response).build();
+	}
+	
+	private Response GenerateddbConnectionGetseqId(StringBuilder seqID) {
+		
+		Responses response      = new Responses();
+		String su = new String();
+		
+
+	    try(Connection connection = datasourceholder.newConnection();
+	            PreparedStatement preparedStatement = connection.prepareStatement("SELECT NEXTVAL('UserSeqID')");
+	              ResultSet resultset = preparedStatement.executeQuery();){
+	    		boolean br = false;
+	    		br = resultset.next();
+	    		
+	    		Date date = new Date();
+	    		SimpleDateFormat dateformat = new SimpleDateFormat("yyyyMMdd");
+	    		String strDate = dateformat.format(date);
+	    		su   = strDate + String.format("%07d", resultset.getInt("nextval"));
+	    		seqID.append(su);
+	    }catch(SQLException e) {
+	    	e.printStackTrace();
+	   
+	    	return GeneralErrorRes(ErrorResponseEnum.E500);
+	    }
+	   
+	    return Response.status(Response.Status.CREATED.getStatusCode()).entity(response).build();
+	    
+	}
+
+	private Response GeneratedbConnectionInsert(RequestBody body, StringBuilder seqID) {
+	    
+	    Responses response      = new Responses();
+	    String strSQLState      = new String();
+	    
+	    int insertCount = 0;
+	   
+	    try(Connection connection = datasourceholder.newConnection();
+	            PreparedStatement preparedStatement = connection.prepareStatement("INSERT INTO T_TOKEN("
+	            + "USER_SEQ_ID,"
+	            + "USER_ID,"
+	            + "DEVICE_TOKEN,"
+	            + "CREATE_DATE,"
+	            + "UPDATE_DATE"
+	            + ") VALUES("
+	            + "?,"
+	            + "?,"
+	            + "?,"
+	            + "current_timestamp,"
+	            + "current_timestamp)");){
+	        
+	        preparedStatement.setString(1, seqID.toString());
+	        preparedStatement.setString(2, body.getId());
+	        preparedStatement.setString(3, body.getToken());
+	        insertCount = preparedStatement.executeUpdate();
+	          
+	          if(insertCount != 1) {
+	     
+	            return GeneralErrorRes(ErrorResponseEnum.E500);
+	          }
+	        }catch(SQLException e) {
+	          
+	          strSQLState = e.getSQLState();
+	          e.printStackTrace();
+	         
+	          if("23505".contentEquals(strSQLState)) {
+	         
+	            return GeneralErrorRes(ErrorResponseEnum.E409POST) ;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the resource returned by the following method call: *GeneralErrorRes*. The resource is returned at line 149. There are other execution paths that don&#39;t close the resource or return it, for example, when DataSourceHolder.newConnection throws an exception. Ensure the object returned by *GeneralErrorRes()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).